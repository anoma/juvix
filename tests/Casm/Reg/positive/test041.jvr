-- Poseidon hash

type PoseidonState {
  poseidon_state : (field, field, field) -> PoseidonState;
}

function poseidon_hash(field, field) : field {
  tmp[0] = alloc poseidon_state (arg[0], arg[1], 2);
  tmp[1] = poseidon tmp[0];
  ret tmp[1].poseidon_state[0];
}

function main() : field {
  tmp[0] = call poseidon_hash (7, 10);
  tmp[1] = fadd tmp[0] 5;
  ret tmp[1];
}
