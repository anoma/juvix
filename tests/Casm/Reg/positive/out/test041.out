960520245973031731151539114206112032362948660473336226878218924279221851535
